### PR TITLE
CLOUDP-251999: [skunkworks] add "atlas" CRD category and short names

### PR DESCRIPTION
**Pull request by s-urbaniak**
This adds a `kubectl get atlas` to be able to retrieve all atlas resources:

```
$ kubectl get atlas -A
NAMESPACE   NAME                                            AGE
ns-search   atlasdeployment.atlas.mongodb.com/search-test   9s

NAMESPACE   NAME                                        NAME
ns-search   atlasproject.atlas.mongodb.com/my-project   atlas-search-test

NAMESPACE   NAME                                                                       AGE
ns-search   atlassearchindexconfig.atlas.mongodb.com/test-search-index-config          8s
ns-search   atlassearchindexconfig.atlas.mongodb.com/test-search-index-config-simple   8s
```

Additionally, this introduces short names for our CRDs:
```
NAME                              SHORTNAMES
atlasbackuppolicies               abp
atlasbackupschedules              abs
atlasdatabaseusers                adu
atlasdatafederations              adf
atlasdeployments                  ad
atlasfederatedauths               afa
atlasprojects                     ap
atlassearchindexconfigs           asic
atlasstreamconnections            asc
atlasstreaminstances              asi
atlasteams                        at
```

This can be used as a shortcut to i.e. list all Atlas Deployments like so:
```
$ kubectl get ad -A
NAMESPACE   NAME          AGE
ns-search   search-test   52s
```

### CODE REVIEW

**Review by josvazg — APPROVED** [review]

**Comment by s-urbaniak** [comment]
spoke with @dan-mckean: we want to write a small PD and scope for this. Once ratified, we can merge this. Putting on hold until then.

**Comment by s-urbaniak** [comment]
as clarified with @dan-mckean we're solving this as a regular Story without PD, scope, spec.

**Comment by s-urbaniak** [comment]
this needs a rebase and adding printcolumns

**Comment by s-urbaniak** [comment]
all done and ready for a final pass.

**Review by roothorp — APPROVED** [review]
Looks fantastic, this feels like a huge usability improvement for a few LoC. Do we need docs for this, or is a release note sufficient?

**Comment by s-urbaniak** [comment]
@roothorp this definitely needs good docs

**Pull request merged**
